src/App.js

```diff
@@ -1,25 +1,193 @@
+  //event listener upon form submition
+  const handleInput = (event) => {
+    setUserInput(event.target.validity.valid ? event.target.value :v);
+    setV('')
+  }
+  const handleTitle =(event) => {
+    setTitle(event.target.value);
+  }
+  const handleBudget = (event) => {
+    setUserBudget(event.target.validity.valid ? event.target.value :v);
+  }
+  const handleSubmit = (event) => {
+    //get the info from userinput STATE
+    event.preventDefault();
+    if (title === "") {
+      Swal.fire(
+        'No name item',
+        'An empty item has been submitted',
+        'info'
+      )
+    }
+    //send it off to firebase using push function
+    const database = getDatabase(firebase);
+    const dbRef = ref(database);
+    const expense = {userInput:userInput, title:title};
+    push(dbRef, expense);
+    setUserInput('');
+    setTitle('');
+  }
+  //event listener for removing {expnese by id}
+  const handleRemove = (expenseId) => {
+    const database = getDatabase(firebase);
+    const dbRef = ref(database, `${expenseId}`);
+    Swal.fire({
+      title: 'Are you sure?',
+      text: "the item will deleted permanently",
+      icon: 'warning',
+      showCancelButton: true,
+      confirmButtonColor: '#3085d6',
+      cancelButtonColor: '#d33',
+      confirmButtonText: 'Yes, delete it!'
+    }).then((result) => {
+      if (result.isConfirmed) {
+        Swal.fire(
+          'Deleted!',
+          'Your file has been deleted.',
+          'success'
+        ).then(remove(dbRef))
+      }
+    })
+    
+  }
+  //fair base linking set up
+  useEffect( () => {
+    const database = getDatabase(firebase);
+    const dbRef = ref(database);
+    //event listenr of firebase for pushing
+    onValue(dbRef, (reponse)=>{
+      const data = reponse.val();
+      const newState = [];
+      for (let key in data) {
+        newState.push(
+          { key:key, 
+            amount: parseInt(data[key]['userInput']),
+            title:data[key]['title'],
+        })};
+      let newTotal = 0;
+      newState.forEach((object) => {
+        newTotal = newTotal + object['amount']
+      })
+      setExpenes(newState);
+      setTotal(newTotal);
+    });
+  },[]);
+  //current we have add expense ✔️
+  //need to rescrict to numbers only✔️
+  //need to make each expense into own div and can be remove "onClick"✔️
+  //need add title into expense✔️
+  //need add budget that reflect in Total amount ?✔️
+  // scretch goal , adding Color reflect the amounts / Total div✔️
+  const color = amount => {
+    if (amount >= 1000){return("expense-cards big-red")}
+  else if(amount >= 600){return("expense-cards gold")}
+  else if(amount >= 450){return("expense-cards raddish")}
+  else if(amount >= 200){return("expense-cards purple")}
+  else if(amount >= 100){return("expense-cards mint")}
+  else if(amount >= 50){return("expense-cards grey")}
+  else if(amount >= 0){return("expense-cards")}
+    //color css for different amount with dynamic changes
```

Comment:
Try to break up into components, might make it easier to navigate!

Reply:
great advice !